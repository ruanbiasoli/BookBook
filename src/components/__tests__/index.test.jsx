import React from "react";
import {
  StyledNavButton,
  NavButton,
  LogoutButton,
} from "../styled/search-page";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Snapshot", () => {
  it("Button from navigation bar - Must be button", () => {
    const StyledNavButtonAction = renderer
      .create(<StyledNavButton onClick={() => {}}>Test</StyledNavButton>)
      .toJSON();
    expect(StyledNavButtonAction).toMatchSnapshot();
  });

  it("Button from navigation bar to change pages - Must be button", () => {
    const NavButtonAction = renderer
      .create(<NavButton onClick={() => {}}>Test</NavButton>)
      .toJSON();
    expect(NavButtonAction).toMatchSnapshot();
  });

  it("Logout button - Must be button", () => {
    const LogoutButtonAction = renderer
      .create(<LogoutButton onClick={() => {}}>Test</LogoutButton>)
      .toJSON();
    expect(LogoutButtonAction).toMatchSnapshot();
  });
});

describe("events", () => {
  it("simulate onclick", () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(
      <StyledNavButton onClick={onClickSpy}>Test</StyledNavButton>
    );
    wrapper.simulate("click");
    expect(onClickSpy).toHaveBeenCalled();
  });

  it("simulate onclick nav button", () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(<NavButton onClick={onClickSpy}>Test</NavButton>);
    wrapper.simulate("click");
    expect(onClickSpy).toHaveBeenCalled();
  });

  it("simulate onclick logout button", () => {
    const onClickSpy = jest.fn();
    const wrapper = shallow(
      <LogoutButton onClick={onClickSpy}>Test</LogoutButton>
    );
    wrapper.simulate("click");
    expect(onClickSpy).toHaveBeenCalled();
  });
});
