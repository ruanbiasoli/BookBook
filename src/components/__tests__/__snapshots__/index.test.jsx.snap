// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Button from navigation bar - Must be button 1`] = `
<button
  className="sc-jSFkmK khqXcO"
  onClick={[Function]}
>
  Test
</button>
`;

exports[`Snapshot Button from navigation bar to change pages - Must be button 1`] = `
<div
  className="sc-pNWxx bhgEAo"
  onClick={[Function]}
>
  Test
</div>
`;

exports[`Snapshot Logout button - Must be button 1`] = `
<button
  className="sc-dIsAE dtWMFF"
  onClick={[Function]}
>
  Test
</button>
`;
